test/lib/resources/registry_test.rb

```diff
@@ -0,0 +1,25 @@
+require 'katello_test_helper'
+
+module Katello
+  module Resources
+    class RegistryTest < ActiveSupport::TestCase
+      before do
+        SETTINGS[:katello][:container_image_registry] = {
+          crane_url: "https://localhost:5000"
+        }
+      end
+
+      def test_pulp3_registry_url
+        pulp_master = FactoryBot.create(:smart_proxy, :default_smart_proxy, :with_pulp3)
+        ::SmartProxy.expects(:pulp_master).at_least_once.returns(pulp_master)
+        assert_equal Registry::RegistryResource.load_class.site,
+                     ::SmartProxy.pulp_master.setting("Pulp3", "content_app_url")
```

Comment:
This may be personal style, but I like to have explicit checks to avoid silently checking `nil` equals `nil`.

```suggestion
                     'http://localhost:24816')
```